parts/linux/cloud-init/artifacts/ubuntu/cse_install_ubuntu.sh

```diff
@@ -303,8 +302,8 @@ installNvidiaDocker() {
     if [ -d "$dst/pkg" ]; then
         if [ -f "$dst/pkg/DEBIAN/control" ]; then
             installed="$(cat "$dst/pkg/DEBIAN/control" | grep Version | cut -d' ' -f 2)"
-            if [ "$version" == "$installed" ]; then
-                echo "skip nvidia-docker2 install, current version $version matches target $target"
+            if [ "$target" == "$installed" ]; then
```

Comment:
$version is empty here, so this always failed, and we pulled from nvidia at runtime.